### DISCUSSION TITLE
scriptibility export is logging a warning and error related to Sage

### DISCUSSION THREAD

**Discussion by bmofkn** [thoughtspot/cs_tools#148] [Q&A] [answered]
When running this command: `cs_tools tools scriptability export --include-types answer --include-system-content ./ --config prod_ts` we get the following Warning and Error related to Sage:

`cs_tools.cli.tools.scriptability._export - WARNING - [b red]unable to get Customers: ERROR`

`cs_tools.cli.tools.scriptability._export - INFO - ERROR d0172390-a19f-4086-973a-a05961cb1d2c Customers Error Code: SAGE_INVALID_QUESTION Incident Id: 6833c0f7-77cd-413a-a5a7-86c80188f41e Error Message: Invalid Sage Question: {0} Could not find the following objects of type LOGICAL_TABLE sent by Sage: 8422558d-d13c-4907-996c-f201ca314bc9`

We have ThoughtSpot Sage experience completely disabled. Is this not a reference to "that Sage"?  

Furthermore, the Answer appears to work and is viewable in the TS UI, but if you export the TML, it simply produces a textfile with the word `null`.  Perhaps this has nothing to do with cs_tools and is related to TS cloud v9.12 which was implemented in our cluster last week (we were not getting this error in cs_tools prior to v9.12)?

**Comment by billdback-ts** [comment, marked as answer]
@bmofkn If I'm understanding correctly, you are saying that you have an answer in TS and can't download the TML (independent of cstools) is that correct?  

Note that Sage is actually a term that's been around for years and originally referred to token search.  I realize it can be confusing, but this message is independent of the new Sage search feature.  

Can you try editing the TML for the answer in TS and looking at the network and console?  cstools won't be able to export if TS has a problem with the saved answer.

**Reply by bmofkn** [reply]
@billdback-ts, thanks for the quick reply.  I cannot edit the TML in the UI.  I get `Current operation is not allowed.` with no TML displayed.  The inspect Network tab shows the following for `/prism/?op=GetEdoc`:

`{
    "data": {
        "Metadata__getEdoc": {
            "zipFile": "UEsDBBQACAgIAEBvxVg<redacted>wAAAJ8BAAAAAA",
            "object": [
                {
                    "edoc": null,
                    "id": "d0172390-a19f-4086-973a-a05961cb1d2c",
                    "name": "Customers",
                    "filename": null,
                    "type": "answer",
                    "statusCode": "ERROR",
                    "errorMessage": "Error Code: SAGE_INVALID_QUESTION Incident Id: 620f91bf-0ef4-4e44-ae76-4f19adc4a9cc\nError Message: Invalid Sage Question: {0} Could not find the following objects of type LOGICAL_TABLE sent by Sage: 8422558d-d13c-4907-996c-f201ca314bc9",
                    "__typename": "EntityDoc"
                }
            ],
            "__typename": "EntityDocResponse"
        }
    }
}`

I'm guessing this is more of a Product issue... I'll file a ticket with support.

**Reply by billdback-ts** [reply]
yeah - that's the best next step.  Thanks.
